lib/fitfree.gi

```diff
@@ -28,7 +28,17 @@ local cache,ffs,pcisom,rest,it,kpc,k,x,ker,r;
 
   pcisom:=ffs.pcisom;
 
-  rest:=RestrictedMapping(ffs.factorhom,U);
+  #rest:=RestrictedMapping(ffs.factorhom,U);
+  if IsPermGroup(U) and AssertionLevel()>1 then
```

Comment:
The commit message says:
>  using `RestrictedMapping` as it will not work for certain matrix group homomorphisms

So, does that mean that there is a bug in `RestrictedMapping`? Or is this more a general problem where e.g. `RestrictedMapping` doesn't have enough information to do something sensible in all cases? Or perhaps a third option?

Also, what exactly is a "matrix group homomorphisms" in this context? A hom with a matrix group as source? Range? Both? Either? And, based on that, I'd have expected that only these kinds of homs would be treated separately here, and the rest handled by `RestrictedMapping` anyway, but that's not the case. I am sure there is a good reason, but what is it?

In any case, IMHO there should be a comment here that briefly explains these things.

Reply:
Changed text.